crates/sui-pg-db/src/db.rs

```diff
@@ -0,0 +1,547 @@
+    ) -> anyhow::Result<Option<ReaderWatermark>> {
+        let watermark: Option<StoredWatermark> = watermarks::table
+            .select(StoredWatermark::as_select())
+            .filter(watermarks::pipeline.eq(pipeline))
+            .first(self)
+            .await
+            .optional()
+            .map_err(anyhow::Error::from)?;
+
+        if let Some(watermark) = watermark {
+            Ok(Some(ReaderWatermark {
+                checkpoint_hi_inclusive: watermark.checkpoint_hi_inclusive as u64,
+                reader_lo: watermark.reader_lo as u64,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn set_committer_watermark(
+        &mut self,
+        pipeline: &'static str,
+        watermark: CommitterWatermark,
+    ) -> anyhow::Result<bool> {
+        // Create a StoredWatermark directly from CommitterWatermark
+        let stored_watermark = StoredWatermark {
+            pipeline: pipeline.to_string(),
+            epoch_hi_inclusive: watermark.epoch_hi_inclusive as i64,
+            checkpoint_hi_inclusive: watermark.checkpoint_hi_inclusive as i64,
+            tx_hi: watermark.tx_hi as i64,
+            timestamp_ms_hi_inclusive: watermark.timestamp_ms_hi_inclusive as i64,
+            reader_lo: 0,
+            pruner_timestamp: NaiveDateTime::UNIX_EPOCH,
+            pruner_hi: 0,
+        };
+
+        use diesel::query_dsl::methods::FilterDsl;
+        Ok(diesel::insert_into(watermarks::table)
+            .values(&stored_watermark)
+            // There is an existing entry, so only write the new `hi` values
+            .on_conflict(watermarks::pipeline)
+            .do_update()
+            .set((
+                watermarks::epoch_hi_inclusive.eq(stored_watermark.epoch_hi_inclusive),
+                watermarks::checkpoint_hi_inclusive.eq(stored_watermark.checkpoint_hi_inclusive),
+                watermarks::tx_hi.eq(stored_watermark.tx_hi),
+                watermarks::timestamp_ms_hi_inclusive
+                    .eq(stored_watermark.timestamp_ms_hi_inclusive),
+            ))
+            .filter(
+                watermarks::checkpoint_hi_inclusive.lt(stored_watermark.checkpoint_hi_inclusive),
+            )
+            .execute(self)
+            .await
+            .map_err(anyhow::Error::from)?
+            > 0)
+    }
+
+    async fn set_reader_watermark(
+        &mut self,
+        pipeline: &'static str,
+        reader_lo: u64,
+    ) -> anyhow::Result<bool> {
+        Ok(diesel::update(watermarks::table)
+            .set((
+                watermarks::reader_lo.eq(reader_lo as i64),
+                watermarks::pruner_timestamp.eq(diesel::dsl::now),
+            ))
+            .filter(watermarks::pipeline.eq(pipeline))
+            .filter(watermarks::reader_lo.lt(reader_lo as i64))
+            .execute(self)
+            .await
+            .map_err(anyhow::Error::from)?
+            > 0)
+    }
+
+    async fn pruner_watermark(
```

Comment:
what's the function order supposed to be here? Keeping getters and setters next to each other is fine, so is keeping all getters together and then all setters together, but it seems like we have a mix. I have a slight preference for the latter, but mostly I care about consistency.